ckanext/harvester_dashboard/logic.py

```diff
@@ -0,0 +1,97 @@
+from ckan import authz, model
+import ckan.plugins.toolkit as tk
+from ckanext.harvest.model import HarvestJob, HarvestSource
+
+HarvestSourceInfo = namedtuple('HarvestSourceInfo', ['organization'])
+OrganizationInfo = namedtuple('OrganizationInfo', ['name', 'title'])
+HarvestJobInfo = namedtuple('HarvestJobInfo', ['last_job'])
+PackageInfo = namedtuple('PackageInfo', ['name'])
+
+log = logging.getLogger(__name__)
+
+
+def get_harvester_job_dict(context, data_dict, harvest_source_ids_for_user):
+    """get latest harvest jobs for each harvest source the user has access to"""
+    harvest_jobs = tk.get_action('harvest_job_list')( {'ignore_auth': True}, {})
+    last_harvest_job_dict = defaultdict(lambda:None)
+    for harvest_job in harvest_jobs:
+        source_id = harvest_job['source_id']
+        user_filter = source_id in harvest_source_ids_for_user
+        job_not_set_filter = not last_harvest_job_dict.get(source_id)
+        if user_filter and job_not_set_filter:
+            last_harvest_job_dict[harvest_job['source_id']] = HarvestJobInfo(
+                last_job=harvest_job
+            )
+    return last_harvest_job_dict
+
+
+def get_harvest_source_name_dict(harvest_source_ids):
+    """get package names for harvest sources, so that urls to the harvesters can be added"""
+    packages = model.Session.query(model.Package)\
+                            .filter(model.Package.id.in_(harvest_source_ids))\
+                            .all()  # noqa
+    harvest_source_name_dict = {package.id: PackageInfo(name=package.name)  # noqa
+                                for package in packages}
+    return harvest_source_name_dict
+
+
+def get_organizations_for_harvest_sources(harvest_source_ids):
+    """gets organizations for harvest_sources, so that it can be decided to
+    which sources a user has access to"""
+    members = model.Session.query(model.Member) \
+                           .filter(model.Member.capacity == 'organization')\
+                           .filter(model.Member.table_name == 'package') \
+                           .filter(model.Member.table_id.in_(harvest_source_ids))\
+                           .all()  # noqa
+    harvest_source_org_dict = {}
+    for member in members:
+        harvest_source_org_dict[member.table_id] = member.group_id
+    return harvest_source_org_dict
+
+
+def get_harvest_source_dict():
+    """gets all active harvest sources"""
+    harvest_sources = model.Session.query(HarvestSource).filter(HarvestSource.active == True).all()  # noqa
+    return {source.id: source for source in harvest_sources}
+
+def get_harvest_organizations_for_user(context, data_dict):
+    """gets all organizations whaere user can administer the harvesters"""
+    organizations_for_user = tk.get_action('organization_list_for_user')(context, data_dict)  # noqa
+    organization_dict_by_id = {}
+    for organization in organizations_for_user:
+        if organization.get('capacity') in ['admin', 'editor']:
```

Comment:
Can you add this to the Readme, that those capacities are required to see the harvest-source on the dashboard?
Maybe it would also be worth it to make this configurable via ini-file.

e.g. `ckanext.harvester_dashboard.roles_permitted_for_harvest_source_access = admin, editor`

Reply:
Thanks for pointing that out. I think I found a better way by directly checking the permissions for the user via check_permission on 'update_harvest_source'